app/models/notify_user/base_notification.rb

```diff
@@ -211,190 +51,97 @@ def delay_time(options)
       end
     end
 
-    # Deliver a single notification across each channel.
-    def self.deliver_channels(notification_id)
-      self.channels.each do |channel_name, options|
-        self.deliver_notification_channel(notification_id, channel_name)
-      end
+    # Check if a hash already exists for that user otherwise create a new one:
+    def generate_unsubscribe_hash
+      NotifyUser::UserHash.where(target: target)
+        .where(type: self.type)
+        .where(active: true)
+        .first_or_create
     end
 
-    # Deliver multiple notifications across each channel as an aggregate message.
-    def self.deliver_channels_aggregated(notifications)
-      self.channels.each do |channel_name, options|
-          if options[:aggregate_per] != false && !unsubscribed_from_channel?(notifications.first.target, channel_name)
-            channel = (channel_name.to_s + "_channel").camelize.constantize
-            channel.deliver_aggregated(notifications, options)
-          end
-      end
+    # Return whether or not the target has unsubscribed from this notification:
+    def target_has_unsubscribed?(channel_name=nil)
+      return Unsubscribe.has_unsubscribed_from?(target, type, group_id, channel_name)
     end
 
-    #deliver to specific channel methods
-
-    # Deliver a single notification to a specific channel.
-    def self.deliver_notification_channel(notification_id, channel_name)
-      notification = self.find(notification_id) # Raise an exception if not found.
-
-      channel_options = channels[channel_name.to_sym]
-      channel = (channel_name.to_s + "_channel").camelize.constantize
-
-      unless notification.user_has_unsubscribed?(channel_name)
-        channel.deliver(notification, channel_options)
-      end
+    def read?
+      read_at.present?
     end
 
-    # Deliver a aggregated notifications to a specific channel.
-    def self.deliver_notifications_channel(notifications, channel_name)
-      channel_options = channels[channel_name.to_sym]
-      channel = (channel_name.to_s + "_channel").camelize.constantize
-
-      #check if user unsubsribed from channel type
-      unless notifications.first.user_has_unsubscribed?(channel_name)
-        channel.deliver_aggregated(notifications, channel_options)
-      end
+    def mark_as_read!
+      self.read_at = Time.zone.now
+      self.save
     end
 
-    # Prepares a single channel for aggregation
-    def self.notify_aggregated_channel(notification_id, channel_name)
-      notification = self.find(notification_id) # Raise an exception if not found.
-
-      # Find any pending notifications with the same type and target, which can all be sent in one message.
-      if self.aggregate_grouping
-        notifications = self.pending_aggregation_by_group_with(notification)
-      else
-        notifications = self.pending_aggregation_with(notification)
-      end
+    def parents_in_group
+      return self.class.none unless self.aggregate_grouping
```

Comment:
Is this preferable to just raising an exception here? I'd say that I'd wanna know sooner rather than later than I'm using a method that makes no sense for the particular class